### ISSUE TITLE
Sync Game Data fails if symlink is already there

### ISSUE THREAD

**Issue by FormBurden** [CryoByte33/steam-deck-utilities#116]
App version: 2.1.4

Steam version: the big stable update that came out on 3-15-23.

After updating Steam yesterday, after the restart, I noticed nearly 60+ games re-downloading cache and shaders started happening, and that made me think... "I bet the symlinks possibly got deleted or reverted because of the update."

I just let everything re-download. Then today, while in Desktop mode I decided to update your app and re-do everything to make sure everything is still working correctly.

When I started to sync game data to my SD-card, I had about 12GBs to forward, which made sense from all those 60+ games re-downloading all that stuff. When I run it, it shows that "None. Everything is synced in this direction", but still have 12GBs to move to the SD-card. So I run it. It runs for a minute, but then errors out and I get this error. Which states "files already exist". As you see in this pic: 
![Screenshot_20230316_190432](https://user-images.githubusercontent.com/30347697/225786523-330f1dfc-c4ed-46f5-bb81-8e7e63251f19.png)

I hit ok, then I get stuck at "moving items, please wait...", which ends up being  at a certain percent (usually 4, or 6, or 8), and will not continue. If I close out of it, it does close. If I run again, it'll be stuck at 0% and never go any where. What you have to do is completely close out of the Deck Utility and restart (from the password prompt). After a few other attempts, it will go beyond 0% but fails out at the next "files already exist", if it does continue. It seems like it can go above 0%, but that only happened like twice out of the 5 times I tried this (before completely quitting the tool).

Not sure how you would want to tackle this, but obviously because of the update the symlinks got deleted or reset, and had to redownload back onto the default location which is on the SD main disk. And re-running the tool it gets the above error. Maybe, if and when the tool is run, if it runs into this issue, it would over-write the files/symlinks, so it can continue the tool. 

I haven't done a "delete game data" yet, because when I tried, it seems like it picks every game on the SD-card and the Deck, and no way to differentiate which is which. Therefore, haven't run that yet to test if the above issue would become resolved or not.

Any other info I could provide, please let me know.

**Comment by CryoByte33** [comment]
I don't think that the update deleted the symlinks, at least mine didn't so it's not so cut-and-dry. Did anyone else get the same behavior?

**Comment by FormBurden** [comment]
Ah dang, hope I wasn't the only one that ran into this issue. 

So after some experimenting, I decided to just "force" run it. Meaning, after I get that message, I close it and keep running. It seems what's happening is once it fails, the script just stops but shows it's still running endlessly, and never progresses the percentage. Once it fails, the specific game that fails gets shuffled down the list, so another can be run the next time. So you do it one by one, and you'll eventually get a game that runs correctly, and the GBs go down, once I run it again because it hit another "fail". So I know it's working.

Hmmm... so I completed everything by using the "delete game data" for the specific games that error out, and even though a tedious process, I got all done, and all symlinks are good. I may have misspoken about symlinks getting deleted. Seemed like the case when "ALL" of my games needed to re-download the shaders. So no idea why I had to do all of that. Anyway, after deleting everything is working correctly. I think the game compdata got corrupted somehow and maybe that is why the issue was happening. I have run into corrupted compdata quite frequently, especially when modding. Things always get broken.

But, even after hitting "ok" when the error prompts come up, the scan gets bugged somehow and freezes. So you have to close the window and do the scan again. That is still an issue. And doing the scan one after another after each time it errors out, you eventually get down to just the corrupted ones only. Maybe we need some kind of detection that if it shows that it has failed (like the symlink file exiting from above) after and "X" amount of times, a prompt will come up to see if we want to delete that compdata (pretty much what "delete game data" does.).

**Comment by accountshayslett** [comment]
I just ran into the exact same problem, step for step, as described, and came here to see if there were any issues being posted about it, and lo and behold, here we are.

**Comment by Schweiger-Jacob** [comment]
I am also having this issue. Has anyone found a solution yet?

**Comment by CryoByte33** [comment]
Sorry about the lack of development, the channel and day jobs have been taking most of my time.

For those currently affected, does "delete game data" and then running the affected game let you sync afterwards?

**Comment by Schweiger-Jacob** [comment]
yes it appears that it does. I deleted all the shaders on the internal storage and now the syncing is working.

**Comment by Perezented** [comment]
Hi, I seem to have ran into this issue. I see it is still open though. I see the specific compatdata/#####/pfx ulr. I figure it would be as easy as identifying what file exists at that rough location and deleting the problem child. I'll dig in some and give an answer if it worked.

**Comment by Perezented** [comment]
Issue #106 seems to be a similar issue if not the same. Same with #177 where that fix would resolve this issue as well. 

-----

So 10090 (COD W@W) exists on my sd card + on the deck nvme ssd
Same with 400 (Portal) and a few others

SD Card:
---
<img width="975" height="517" alt="Image" src="https://github.com/user-attachments/assets/a57d04d8-a344-4598-99d6-073bfc619860" />


NVME SSD:
---
<img width="953" height="228" alt="Image" src="https://github.com/user-attachments/assets/f996aaa2-5f25-44fa-a97a-21f526e201cc" />


NVME SSD seems to have less data on it too.  I used the tool to `Clean`/`Delete Game Data` and it removed both compatdata folders for COD W@W. 

I did a deletion of the smaller [file/folder] count after comparing the two folders on both ends, then ran the `Sync Game Data` with success. I did run into the issue about 5 total times

<img width="2244" height="1272" alt="Image" src="https://github.com/user-attachments/assets/fa15e3fc-a063-4113-90b6-ba487dc99fc9" />


I haven't confirmed or denied if it is the correct data- hoping cloud data kicks in
